### ISSUE TITLE
Do not link with -ffast-math

### ISSUE THREAD

**Issue by gcode-importer** [uclouvain/openjpeg#488]
Originally reported on Google Code with ID 488
```
Linking shared libraries with -ffast-math is dangerous.  It causes the SSE mxcsr flags
to be set on library *load*, which can cause various issues for programs the unsuspectingly
load such a library.  See:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=55522
https://bugzilla.redhat.com/show_bug.cgi?id=1127544

for some discussion, examples.

Unfortunately, CMAKE_C_FLAGS are also used during link, so the current:

   # Do not use ffast-math for all build, it would produce incorrect results,
only set for release:
   SET(CMAKE_C_FLAGS_RELEASE "-ffast-math ${CMAKE_C_FLAGS_RELEASE}")

triggers this.

According to http://public.kitware.com/pipermail/cmake/2015-April/060479.html the preferred
solution would be something like:

target_compile_options(openjpeg PRIVATE
  $<$<CONFIG:Release>:-ffast-math>)

The attached patches appears to achieve this (for 1.5.1 and trunk).

Another wrinkle on this is that -DCMAKE_BUILD_TYPE=RelWithDebInfo or MinSizeRel will
also cause -ffast-math not to be used at all.  Not sure if that is an issue or not.

```

Reported by `orion@cora.nwra.com` on 2015-04-28 21:43:08

<hr>
 * *Attachment: [openjpeg-fast-math.patch](https://storage.googleapis.com/google-code-attachments/openjpeg/issue-488/comment-0/openjpeg-fast-math.patch)*
 * *Attachment: [openjpeg2-fast-math.patch](https://storage.googleapis.com/google-code-attachments/openjpeg/issue-488/comment-0/openjpeg2-fast-math.patch)*

**Comment by rdieter** [comment]
I can take care of importing the proposed patches

**Comment by rdieter** [comment]
Hrm, apparently I don't have commit access yet :(

**Comment by mayeut** [comment]
Done on master

**Comment by mayeut** [comment]
This requires CMake 2.8.12
Minimum CMake version was 2.8.2 until now.
Either disable -ffast-math before 2.8.12 or bump minimum CMake version to 2.8.12 ?

**Comment by mayeut** [comment]
@detonin,

Feel free to close if the implemented solution is OK.
> disable -ffast-math before 2.8.12

**Comment by detonin** [comment]
Implemented solution is ok. Thanks. Closing.
Master will be merged in 2.1 branch so #553 shall be closed too.
